src/CustomAuthenticationProvider.ts

```diff
@@ -47,6 +48,9 @@ export class CustomAuthenticationProvider implements AuthenticationProvider {
 				if (accessToken) {
 					resolve(accessToken);
 				} else {
+					if (!error) {
+						error = new GraphError(-1, "Access token cannot be undefined or empty.");
```

Comment:
We really should have a GraphClientError class since GraphError is modeled around OData errors. A GraphClientError object should provide actionable information since we typically know what caused an error. Ideally, we provide URL with an aka.ms link to information that the developer needs to resolve the issue.

If we know how the access token may have gotten in the undefined or empty state, we should provide guidance on how to fix this.

Reply:
Added a GraphClientError class